### ISSUE TITLE
File/Media Service

### ISSUE THREAD

**Issue by jenniferplusplus** [Letterbook/Letterbook#76]

**Comment by runewake2** [comment]
We’ll also probably want to be able to remove image metadata, location data etc from images that get uploaded as part of this. Some way to scrub out personal EXIF data automatically would be great. For privacy reasons I don't think we'll want to reuse images directly.

**Comment by jenniferplusplus** [comment]
Yeah, definitely. I would like if we could do that selectively. I personally like to have camera settings, for example. Or at least look for and preserve descriptive text before stripping metadata. But for sure we should redact or remove it before serving the image anywhere.

And yes, doing some pre-processing to improve cost and performance is also important.

Did you want to make a start on this service?

**Comment by runewake2** [comment]
I'm slowly getting more time to work on this. Happy to poke around this weekend and try to put something together.

**Comment by runewake2** [comment]
Thinking about this it is probably a good place to design #40. Having a series of pipelines to process, remux, compress etc media in a way that's configurable by the server admin could be handy.

**Comment by jenniferplusplus** [comment]
I really don't have any good ideas (or even bad ideas) for how to build out a plugin architecture. I worry it's premature at this point. But, if you can put something together, I'm more than happy to review it.
